src/App/frontend/src/features/datamodel/DataModelsProvider.tsx

```diff
@@ -156,11 +160,13 @@ function DataModelsLoader() {
   const layouts = useLayouts();
   const defaultDataType = useCurrentDataModelName();
   const isStateless = useApplicationMetadata().isStatelessApp;
+  const queryClient = useQueryClient();
+  const { instanceOwnerPartyId, instanceGuid } = useInstanceDataQueryArgs();
 
-  const { data: dataElements, isFetching } = useInstanceDataQuery({
-    enabled: !isStateless,
-    select: (data) => data.data.map((element) => [element.dataType, element.locked] as const),
-  });
+  const dataElements =
+    queryClient.getQueryData(instanceQueries.instanceData({ instanceOwnerPartyId, instanceGuid }).queryKey)?.data ??
+    window.AltinnAppData?.instance?.data ??
+    emptyArray;
```

Comment:
Ved å bruke `queryClient.getQueryData` i stedet for `useQuery` vil verdien her ikke oppdatere seg når dataen i cachen oppdateres. Er det meningen? Hvis det er meningen å bruke initiell data og unngå at requestet blir fyrt kan du fortsatt bruke `useQuery`, men sette `initialData` og `staleTime` > 0. Da vil cachen behandle `initialData` som fersk data og ikke refetche før `staleTime` har gått ut. De den deklarative måten å sette initiell data på her: https://tanstack.com/query/latest/docs/framework/react/guides/initial-query-data#initial-data-from-the-cache-with-initialdataupdatedat

Reply:
Vi kan også vurdere å bruke `usePrefetchQuery` med `initialData` og `staleTime` på ett samlet sted.